src/game/ZoneArea.cxx

```diff
@@ -0,0 +1,153 @@
+#include "../engine/Engine.hxx"
+
+void mergeZoneAreas(ZoneArea &mainZone, ZoneArea &toBeMerged)
+{
+  mainZone.m_zoneNodes.insert(mainZone.m_zoneNodes.end(), toBeMerged.m_zoneNodes.begin(), toBeMerged.m_zoneNodes.end());
+  mainZone.m_hasPower |= toBeMerged.m_hasPower;
+  mainZone.m_hasWater |= toBeMerged.m_hasWater;
+  mainZone.xmin = std::min(mainZone.xmin, toBeMerged.xmin);
+  mainZone.xmax = std::max(mainZone.xmax, toBeMerged.xmax);
+  mainZone.ymin = std::min(mainZone.ymin, toBeMerged.ymin);
+  mainZone.ymax = std::max(mainZone.ymax, toBeMerged.ymax);
+};
+
+ZoneArea::ZoneArea(ZoneNode zoneNode)
+    : m_zoneNodes{zoneNode}, xmin(std::max(0, zoneNode.coordinate.x - 1)), m_zoneType(zoneNode.zoneType),
+      m_zoneDensity(zoneNode.zoneDensity), xmax(std::min(Settings::instance().mapSize, zoneNode.coordinate.x + 1)),
+      ymin(std::max(0, zoneNode.coordinate.y - 1)), ymax(std::min(Settings::instance().mapSize, zoneNode.coordinate.y + 1))
+{
+}
+
+void ZoneArea::spawnBuildings()
+{
+
+  constexpr int amountOfBuildingsToSpawn = 5;
+  auto &randomizer = Randomizer::instance();
+  // shuffle mapNodes so placement of building looks random
+  randomizer.shuffle(m_zoneNodes.begin(), m_zoneNodes.end());
+
+  int buildingsSpawned = 0;
+
+  // pick every single zone node we have
+  for (auto &node : m_zoneNodes)
+  {
+    if (node.occupied)
+    {
+      continue;
+    }
+
+    if (buildingsSpawned >= amountOfBuildingsToSpawn)
+    {
+      break;
+    }
+    // get the maximum size we can spawn at this node
+    TileSize maxTileSize = {getMaximumTileSize(node.coordinate).width, getMaximumTileSize(node.coordinate).height};
+    std::string buildingTileID = TileManager::instance().getRandomTileIDForZoneWithRandomSize(getZone(), getZoneDensity(), maxTileSize);
+
+    // place the building
+    std::vector targetObjectNodes = Engine::instance().map->getObjectCoords(node.coordinate, buildingTileID);
+    Engine::instance().setTileIDOfNode(targetObjectNodes.begin(), targetObjectNodes.end(), buildingTileID, false);
+    buildingsSpawned++;
+  }
+}
+
+bool ZoneArea::isPartOfZone(Point coordinate) const
+{
+  return m_zoneNodes.end() != std::find_if(m_zoneNodes.begin(), m_zoneNodes.end(),
+                                           [&coordinate](const ZoneNode &node) { return node.coordinate == coordinate; });
+}
+
+bool ZoneArea::isVacant() const
+{
+  return m_zoneNodes.end() !=
+         std::find_if(m_zoneNodes.begin(), m_zoneNodes.end(), [](const ZoneNode &node) { return node.occupied == false; });
+}
+
+bool ZoneArea::isNeighborOfZone(Point coordinate) const
+{
+  for (ZoneNode node : m_zoneNodes)
+  {
+    if ((node.coordinate.x == coordinate.x) && (std::abs(node.coordinate.y - coordinate.y) <= 1) ||
+        (node.coordinate.y == coordinate.y) && (std::abs(node.coordinate.x - coordinate.x) <= 1))
+    {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+TileSize ZoneArea::getMaximumTileSize(Point originPoint)
+{
+  TileSize possibleSize;
```

Comment:
You count on default width and height values. Nice surprise when someone changes TileSize class.

Reply:
TileSize is intended for specifying the "tileSize" (meaning how many tiles a building uses, or how big it is). This is exactly what is for, i  don't understand how or why anyone would change this, because it would change the usecase for it?
If you have a better name for the struct or for the members, we could change it.
But this is used correctly

Reply:
So what I said is following. You are implicit counting on that , that dimensions of TileSize are set to 1. If this changes in future, will introduce bug, which is not so easy to track. Safer is to initialize it on correct dimension.

Reply:
But a building can never use a lower tilesize than 1x1 so no one should ever change the initial values. If it was changed it would cause a lot of issues, not only here.
But I initialize it with {1, 1} now